src/staging.c

```diff
@@ -184,6 +194,17 @@ enum swupd_code do_staging(struct file *file, struct manifest *MoM)
 			ret = SWUPD_COULDNT_RENAME_DIR;
 			goto out;
 		}
+
+		/* add the directory name to the staging member to indicate the
+		 * staging was successful */
+		free_string(&file->staging);
+		string_or_die(&file->staging, "%s%s/%s", path_prefix, rel_dir, base);
```

Comment:
I didn't like this approach. We use the staging variable to keep the name of a temporary location of the file and this one is already in it's final destination. The problem you reported isn't a problem on update. It looks like we should do a clean up on bundle add to avoid doing this here. Take a look at the update_loop() on update.c.

Reply:
ok